### PR TITLE
[AAP-53741] Add RequestAuditInfoFilter for source IP and user agent logging

### PR DESCRIPTION
**Pull request by john-westcott-iv**
## Description

- **What is being changed?** Adds a new `RequestAuditInfoFilter` log filter in `ansible_base/lib/logging/filters/request_audit_info.py` that extracts source IP and user agent from HTTP requests and adds them to log records as `record.source_ip` and `record.user_agent`.

- **Why is this change needed?** Authentication audit logs should include client identification information (IP address and user agent) to help with security auditing, troubleshooting, and compliance requirements.

- **How does this change address the issue?** The new filter:
  - Extracts `source_ip` from `X-Forwarded-For` header (first IP if multiple) or falls back to `REMOTE_ADDR`
  - Extracts `user_agent` from `HTTP_USER_AGENT` header
  - Sets empty string defaults when no request context is available
  - Always returns `True` so messages are logged regardless of request availability

## Type of Change
- [x] New feature (non-breaking change which adds functionality)
- [x] Test update

## Self-Review Checklist
- [x] I have performed a self-review of my code
- [x] I have added relevant comments to complex code sections
- [x] I have considered the security impact of these changes
- [x] I have considered performance implications
- [x] I have thought about error handling and edge cases
- [x] I have tested the changes in my local environment

### Prerequisites
- tox and tox-docker installed

### Steps to Test
1. Run `tox -e py311 -- test_app/tests/lib/logging/filters/test_request_audit_info.py -v`
2. All tests should pass

### Expected Results
- Tests verify source IP extraction from REMOTE_ADDR and X-Forwarded-For
- Tests verify user agent extraction
- Tests verify empty defaults when no request context
- Integration test verifies filter works in logging chain

### Required Actions
- [x] Requires downstream repository changes
  - aap-gateway: PR #1154 adds handlers/formatters that use this filter

---
**Note:** This PR was developed with assistance from Claude AI assistant.

### CODE REVIEW

File: ansible_base/lib/logging/filters/request_audit_info.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        It extracts the source IP and user agent from the request.
+        It will always return True, so that the message is always logged,
+        even if there is no request.
+        """
+
+        # Always set defaults so we can use them in the logging formatter.
+        record.source_ip = ""
+        record.user_agent = ""
+
+        request = getattr(thread_local, "request", None)
```
Comment (bhavenst) [inline ansible_base/lib/logging/filters/request_audit_info.py:19]: I suppose this general structure could be abstractified into some kind of nonblocking request logging filter with plugins to reduce the overlap with RequestIdFilter, but no big deal there.  Maybe if a 3rd one comes along..

**Review by bhavenst — APPROVED** [review]

File: ansible_base/lib/logging/filters/request_audit_info.py (hunk 1 above)
Comment (john-westcott-iv) [reply ansible_base/lib/logging/filters/request_audit_info.py:19]: We could do something like that. We would just potentially be extracting extra information for logs that don't need it. So depending on how big a "generic" one became it could start causing issues. It's honestly be nice if some of this info was "just provided" as context to the logger out of the gate.
